ale/util.py

```diff
@@ -506,3 +501,303 @@ def query_kernel_pool(matchstr="*"):
     svals = [duckpool(v) for v in svars]
     return dict(zip(svars, svals))
 
+
+def read_pvl(path, use_jank=False):
+    """
+    Syntax sugar, used to load a pvl object file from path
+
+    Parameters
+    ----------
+
+    path : str
+           Path to Pvl file
+
+    use_jank : bool
+               If true, uses faster but less reliable JBFPvlParser, else uses standard PVL parser.
+
+    """
+    with open(path) as f:
+        preftext = f.read().replace('EndGroup', 'End_Group').replace("EndObject", "End_Object")
+        if use_jank:
+            pvlprefs = JBFPvlParser(open(path).read())
+        else:
+            pvlprefs = pvl.loads(preftext)
+
+    return pvlprefs
+
+
+def get_isis_mission_translations(isis3_data):
+    """
+    Use ISIS translation files and return a lookup table.
+
+    Parameters
+    ----------
+
+    isis3_data : str
+                 path to $ISIS3DATA
+
+    Returns
+    -------
+
+    : dict
+      Dictionary mapping label mission strings to ISIS3 mission strings
+
+    """
+    mission_translation_file = read_pvl(os.path.join(isis3_data, "base", "translations", "MissionName2DataDir.trn"))
+    # For some reason this file takes the form [value, key] for mission name -> data dir
+    lookup = [l[::-1] for l in mission_translation_file["MissionName"].getlist("Translation")]
+    return dict(lookup)
+
+
+def JBFPvlParser(lines):
+    """
+    Janky But Faster PVL Parser(TM)
+
+    Only really supports ISIS's Kernel DB files. This is because KernelDB files are sometimes very large for smithed kernels.
+    This should bring the parsing time for those DB files from minutes to seconds.
+    Still needs nested object/group support and it should be able to read most PVL files.
+
+    Parameters
+    ----------
+
+    lines : str
+          string body of PVL file.
+
+    Returns : PVLModule
+              object representing the parsed PVL
+
+
+    """
+    def JBFKeywordParser(lines):
```

Comment:
This looks like it only works with a single line. Why pass in a list?

Reply:
Keywords can make up multiple lines. Arrays and long wrapping keywords are two common examples.